From f1778f34e0848bdcbe471c91f6ee7f9146e2d9f8 Mon Sep 17 00:00:00 2001
From: Assam Boudjelthia <assam.boudjelthia@qt.io>
Date: Wed, 10 May 2023 11:23:20 +0300
Subject: [PATCH] Android: fix QDir::entryInfoList() with content URIs

Correctly return only the fileName portion from
AndroidContentFileEngineIterator::currentFileName(); which
QDirIteratorPrivate::matchesFilters() expects when looking for matches
of the file pattern provided to the QDirIterator.

Fixes: QTBUG-112738
Pick-to: 6.5 6.2 5.15
Change-Id: I98dba2df014b27f33cd7e54fab3ad2de8c7c1750
Reviewed-by: Ville Voutilainen <ville.voutilainen@qt.io>
---

diff --git a/src/plugins/platforms/android/androidcontentfileengine.cpp b/src/plugins/platforms/android/androidcontentfileengine.cpp
index ed2238c..c86bbea 100644
--- a/src/plugins/platforms/android/androidcontentfileengine.cpp
+++ b/src/plugins/platforms/android/androidcontentfileengine.cpp
@@ -351,14 +351,14 @@
 {
     if (m_index < 0 || m_index > m_files.size())
         return QString();
-    // Returns a full path since contstructing a content path from the file name
-    // and a tree URI only will not point to a valid file URI.
-    return m_files.at(m_index)->uri().toString();
+    return m_files.at(m_index)->name();
 }
 
 QString AndroidContentFileEngineIterator::currentFilePath() const
 {
-    return currentFileName();
+    if (m_index < 0 || m_index > m_files.size())
+        return QString();
+    return m_files.at(m_index)->uri().toString();
 }
 
 // Start of Cursor
